src/settings.rs

```diff
@@ -78,6 +90,7 @@ pub struct Settings {
     pub cluster: ClusterConfig,
     #[serde(default = "default_telemetry_disabled")]
     pub telemetry_disabled: bool,
+    pub tls_config: TlsConfig,
```

Comment:
Ditto. (E.g., `#[serde(default = ...)]` or `Option<TlsConfig>`?)

Reply:
I would prefer `Option<TlsConfig>`

Reply:
@ffuugoo good catch, btw. It could cause panic if somebody would try to upgrade without updating config (if custom config is provided). And our backward compatibility test don't check this currently